### PR TITLE
Pypy3

### PR DESCRIPTION
**Pull request by mattip**

### CODE REVIEW

**Comment by mattip** [comment]
CI fails to build, [missing `PyUnicode_EncodeCodePage`](https://github.com/mattip/pywin32/runs/6205132302?check_suite_focus=true#step:7:340) in `win32/src/PyUnicode.cpp`

**Comment by mattip** [comment]
Close/reopen to see where we stand with this now

**Comment by mattip** [comment]
Close/reopen to see where we stand with this now

File: win32/src/PyUnicode.cpp [hunk 1]
```diff
@@ -103,12 +103,17 @@ BOOL PyWinObject_AsString(PyObject *stringObject, char **pResult, BOOL bNoneOK /
...
     if (PyUnicode_Check(stringObject)) {
+#if defined(PYPY_VERSION_NUM) && PYPY_VERSION_NUM < 0x07030b00 
+        stringObject = tempObject =
+            PyUnicode_EncodeMBCS(PyUnicode_AS_UNICODE(stringObject), PyUnicode_GET_SIZE(stringObject), NULL);
+#else
         // PyUnicode_EncodeMBCS was removed in Py 3.11.
         PyObject *unicode = PyUnicode_FromWideChar(PyUnicode_AS_UNICODE(stringObject), -1);
         if (unicode == NULL)
             return FALSE;
         stringObject = tempObject = PyUnicode_EncodeCodePage(CP_ACP, unicode, NULL);
```
Comment (Avasam) [inline win32/src/PyUnicode.cpp]: So you know, this was removed in https://github.com/mhammond/pywin32/pull/1860 , you might wanna pull upstream again

File: win32/Lib/pywin32_testutil.py [hunk 2]
```diff
@@ -62,28 +65,30 @@ def countTestCases(self):
...
+            gc.collect()
+            lost_i = _GetInterfaceCount() - ni
+            lost_g = _GetGatewayCount() - ng
+            if lost_i or lost_g:
+                msg = "%d interface objects and %d gateway objects leaked" % (
+                    lost_i,
+                    lost_g,
+                )
+                exc = AssertionError(msg)
+                result.addFailure(self.real_test, (exc.__class__, exc, None))
```
Comment (Avasam) [inline win32/Lib/pywin32_testutil.py:91]: You're voluntarily causing leaks in tests here. At this point, better just disable the test entirely on PyPy

File: setup.py [hunk 3]
```diff
@@ -1004,7 +1017,7 @@ def get_ext_filename(self, name):
             return "win32\\pythonservice" + extra_exe
         elif name.endswith("pythonwin.Pythonwin"):
             return "pythonwin\\Pythonwin" + extra_exe
-        return name.replace(".", "\\") + ".pyd"
+        return name.replace(".", "\\") + EXT_SUFFIX
```
Comment (Avasam) [inline setup.py]: This isn't equivalent
```py
>>> import sysconfig
>>> sysconfig.get_config_var("EXT_SUFFIX")
'.cp39-win_amd64.pyd'
```

Comment (mattip) [reply setup.py]: fixed in #2

File: win32/src/PyUnicode.cpp (hunk 1 above)
Comment (mattip) [reply win32/src/PyUnicode.cpp]: thanks, see #2

**Pull request closed without merging**